src/orion/storage/legacy.py

```diff
@@ -135,10 +170,132 @@ def retrieve_result(self, trial, results_file=None, **kwargs):
 
         return trial
 
-    def update_trial(self, trial: Trial, where=None, **kwargs) -> Trial:
+    def get_trial(self, trial=None, uid=None):
+        """See :func:`~orion.storage.BaseStorageProtocol.get_trial`"""
+        if uid is None:
+            uid = trial.id
+
+        result = self._db.read('trials', {'_id': uid})
+        if not result:
+            return None
+
+        return Trial(**result[0])
+
+    def _update_trial(self, trial: Trial, where=None, **kwargs) -> Trial:
         """See :func:`~orion.storage.BaseStorageProtocol.update_trial`"""
         if where is None:
             where = dict()
 
         where['_id'] = trial.id
         return self._db.write('trials', data=kwargs, query=where)
+
+    def fetch_lost_trials(self, experiment):
+        """See :func:`~orion.storage.BaseStorageProtocol.fetch_lost_trials`"""
+        # TODO: Configure this
+        threshold = datetime.datetime.utcnow() - datetime.timedelta(seconds=60 * 2)
+        lte_comparison = {'$lte': threshold}
+        query = {
+            'experiment': experiment._id,
+            'status': 'reserved',
+            'heartbeat': lte_comparison
+        }
+
+        return self._fetch_trials(query)
+
+    def push_trial_results(self, trial):
+        """See :func:`~orion.storage.BaseStorageProtocol.push_trial_results`"""
+        return self._update_trial(trial, **trial.to_dict(), where={'_id': trial.id})
+
+    def set_trial_status(self, trial, status, heartbeat=None):
+        """See :func:`~orion.storage.BaseStorageProtocol.set_trial_status`"""
```

Comment:
This method needs more documentation. What happens if trial.status is not the same as in the DB?

Reply:
shouldn't we just remove the query status=current_status.
The only time where there is a concurrent status update is when the monitor is updating the status. I do not think that it is that important now

Reply:
It's true that the most important point for this which was in `experiment.reserve_trial` is not needed anymore. It seams to be a reasonable safeguard to keep nevertheless. There should not be 2 process messing with the status at the same time, so if this ever happens I think it should not go unnoticed.